hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/EntityManagerFactoryBuilderImpl.java

```diff
@@ -196,8 +196,19 @@ private EntityManagerFactoryBuilderImpl(
 			integrationSettings = Collections.emptyMap();
 		}
 
+		Map mergedIntegrationSettings = null;
+		Properties properties = persistenceUnit.getProperties();
+		if ( properties != null ) {
+			mergedIntegrationSettings = new HashMap( persistenceUnit.getProperties() );
```

Comment:
I think at least we can reuse `properties` at the above statement to improve code readability a little bit. 
```
if (properties != null ) {
    mergedIntegrationSettings = new HashMap( properties );
    ... ...
}
```

Reply:
+1 good catch

Reply:
@NathanQingyangXu do you want to create a PR for that since it was your catch?

Reply:
There is another subtle bug that if the passed 'integrationSettings' is null the entries from PU might be added to `Collections.emptyMap()` ending up with `UnsupportedOperationException`.
Yeah, I am gonna create a PR to fix all such issues.

Reply:
PR created at: https://github.com/hibernate/hibernate-orm/pull/3246. It might be a little bit aggressive, but I do think we can improve code readability besides fixing a subtle potential bug.